frontend/models/product-list/server.ts

```diff
@@ -401,3 +424,27 @@ function createPublicAlgoliaKey(appId: string, apiKey: string): string {
    });
    return publicKey;
 }
+
+async function findDevicesWithProducts(devices: string[]) {
+   return timeAsync('algolia.findDescendantDevicesWithProducts', async () => {
+      const client = algoliasearch(
+         ALGOLIA_APP_ID,
+         ALGOLIA_API_KEY,
+         CLIENT_OPTIONS
+      );
+      const index = client.initIndex(ALGOLIA_PRODUCT_INDEX_NAME);
+      const deviceDisjunctiveFilters = devices
+         .map((device) => `device:"${escapeFilterValue(device)}"`)
+         .join(' OR ');
+      const deviceFilterSuffix = deviceDisjunctiveFilters
+         ? ` AND ${deviceDisjunctiveFilters}`
```

Comment:
I would expect this would need to be wrapped in parenthesis. Otherwise will end up with `public=1 AND device:X OR device:Y OR device:Z`

Reply:
Good catch

Reply:
https://github.com/iFixit/react-commerce/pull/1252/commits/7df5256696bc5596c6bcdc0a2abc1408854ad754